optimum/intel/openvino/quantization.py

```diff
@@ -330,25 +328,90 @@ def _quantize_torchmodel(
             model_type=model_type,
         )
 
-        if self.task == "text-generation":
-            onnx_config = onnx_config_class(self.model.config, use_past=self.model.config.use_cache)
+        if task == "text-generation":
+            onnx_config = onnx_config_class(model.config, use_past=model.config.use_cache)
         else:
-            onnx_config = onnx_config_class(self.model.config)
+            onnx_config = onnx_config_class(model.config)
 
-        compressed_model.eval()
-        num_parameters = compressed_model.num_parameters()
-        save_as_external_data = use_external_data_format(num_parameters) or quantization_config.save_onnx_model
-        f = io.BytesIO() if not save_as_external_data else save_directory / ONNX_WEIGHTS_NAME
+        onnx_path = save_directory / ONNX_WEIGHTS_NAME
 
-        # Export the compressed model to the ONNX format
+        # Export the model to the ONNX format
         opset = min(onnx_config.DEFAULT_ONNX_OPSET, MAX_ONNX_OPSET)
-        opset = opset if not quantization_config.save_onnx_model else max(opset, MIN_ONNX_QDQ_OPSET)
-        _onnx_export_nncf_model(compressed_model, onnx_config, f, opset)
+        opset = max(opset, MIN_ONNX_QDQ_OPSET)
+        export(
+            model=compressed_model,
+            config=onnx_config,
+            opset=opset,
+            output=onnx_path,
+        )
 
         # Load and save the compressed model
-        model = core.read_model(f) if save_as_external_data else core.read_model(f.getvalue(), b"")
+        model = core.read_model(onnx_path)
         self._save_pretrained(model, output_path)
         quantization_config.save_pretrained(save_directory)
+        if not quantization_config.save_onnx_model:
+            os.remove(onnx_path)
+            try:
+                os.remove(f"{onnx_path}_data")
+            except FileNotFoundError:
+                pass
+
+    def compress_weights(
```

Comment:
I would prefer to add this in the `.quantize()` method, enabling it with an argument such as `weights_only` or `disable_activation`

Reply:
Also in the case where a user gives a config should we verify the activations `ignored_scopes` value ? Otherwise quantization could be applied on the activations when this is unpexpected no?

Reply:
There are several works on the mixing sparsity + quantization to compress models in the post-training mode. At some point, we can extend the implementation.

Reply:
> Also in the case where a user gives a config should we verify the activations `ignored_scopes` value ? Otherwise quantization could be applied on the activations when this is unpexpected no?

You are right here, for now, it is possible to hack this method using quantization_config. As I mentioned we will pug a new API in the future. I added `quantization_config` for experimenting purposes mostly, but no problem to delete it from here. What do you think?

Reply:
> It may work. The thing is that in NNCF it is supposed to be a separate API call (implementation is in progress). @echarlaix, what are the benefits of having all in one API, in your opinion?


Currently we try to unify the quantization API of the different optimum subpackages in order for users to be able to easily transition from one to another (`ORTQuantizer` -> `OVQuantizer`)
We plan to create 
Having a close behavior among the quantization API of the different optimum subpackages makes it easier for users 


We want the quantization API of the different optimum subpackages to have a behave as close as possible in order 


so that it's easy for users to transition from one subpackageto the other, 
Currently we try to unify the quantization API of the different optimum subpackages in order for users to be able to easily transition from one to another (`ORTQuantizer` -> `OVQuantizer`)

Reply:
> There are several works on the mixing sparsity + quantization to compress models in the post-training mode. At some point, we can extend the implementation.

Super interesting ! Happy to help on the integration if needed

Reply:
> > Also in the case where a user gives a config should we verify the activations `ignored_scopes` value ? Otherwise quantization could be applied on the activations when this is unpexpected no?
> 
> You are right here, for now, it is possible to hack this method using quantization_config. As I mentioned we will pug a new API in the future. I added `quantization_config` for experimenting purposes mostly, but no problem to delete it from here. What do you think?

I agree with the current integration, it makes sense to leave the users the possibility to give a config (as currently the case for weights and activations quantization). Something that we can do is overwritte the config by adding the default patterns when missing and add a warning, what do you think ? Also after a quick look to the repo, I can see that all the examples config are for QAT / JPQD and our PTQ documentation / examples all use the default configuration, so it's not as likely as I thought originally, happy to do it in a following PR if easier for you

Reply:
Done in https://github.com/huggingface/optimum-intel/pull/359